### PR TITLE
Add a Makefile

### PR DESCRIPTION
**Pull request by oleorhagen**
And a `version` command to the `mender-shell`

### CODE REVIEW

File: config/version.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Northern.tech AS
```
Comment (merlin-northern) [inline config/version.go:1]: perhaps 2020?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	@echo "-- checking for dead code"
+	@deadcode -ignore version.go:Version
+
+govarcheck:
+	@echo "-- checking with varcheck"
+	@varcheck ./...
+
+gocyclo:
+	@echo "-- checking cyclometric complexity > $(GOCYCLO)"
+	@gocyclo -over $(GOCYCLO) $(PKGFILES_notest)
```
Comment (merlin-northern) [inline Makefile:88]: we define $GO so maybe lets define gocyclo and the above two as well?

**Review by merlin-northern — DISMISSED** [review]
lgtm. small comments, I trust you :)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,120 @@
+DESTDIR ?= /
+prefix ?= $(DESTDIR)
+bindir=/usr/bin
+datadir ?= /usr/share
+sysconfdir ?= /etc
+systemd_unitdir ?= /lib/systemd
+docexamplesdir ?= /usr/share/doc/mender-client/examples
```
Comment (kacf) [inline Makefile:7]: Copy paste error.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	else \
+		mv coverage-tmp.txt coverage.txt; \
+	fi
+	@rm -f coverage-tmp.txt coverage-missing-subtests.txt
+
+.PHONY: build
+.PHONY: clean
+.PHONY: get-tools
+.PHONY: test
+.PHONY: check
```
Comment (kacf) [inline Makefile:111]: `test` and `check` targets are missing. Just follow what the client does, I suppose?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+install-bin: mender-shell
+	@install -m 755 -d $(prefix)$(bindir)
+	@install -m 755 mender-shell $(prefix)$(bindir)/
+
+install-systemd:
+	@install -m 755 -d $(prefix)$(systemd_unitdir)/system
+	@install -m 0644 support/mender-shell.service $(prefix)$(systemd_unitdir)/system/
+
+uninstall: uninstall-bin uninstall-systemd
```
Comment (kacf) [inline Makefile:61]: Both dependencies don't exist.

File: config/version.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+func VersionString() string {
+	if Version != "" {
+		return Version
+	}
+	return "unknown"
+}
+
+func ShowVersionCLI(ctx *cli.Context) error {
+	fmt.Printf(ShowVersion())
```
Comment (kacf) [inline config/version.go:36]: Should this be `Println` instead?

**Review by kacf — CHANGES_REQUESTED** [review]

**Review by kacf — APPROVED** [review]

**Pull request merged**